src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableAnalyzer.cs

```diff
@@ -52,189 +55,205 @@ public static void CompilationStart(CompilationStartAnalysisContext context)
 				return;
 			}
 
+			/* exclusively checking the existence of the Enumerable.Append method because 
+			   Enumerable.Append and Enumerable.Prepend were released at the same time */
 			var hasEnumerableAppendMethod = HasEnumerableAppendMethod(context.Compilation);
 			var cache = new FileDetailCache();
 
 			context.RegisterSyntaxNodeAction(c => Analyze(c, cache, hasEnumerableAppendMethod), SyntaxKind.SimpleMemberAccessExpression);
 		}
 
-		static bool ContainsSingleElement (ExpressionSyntax? e)
-		{
-			switch (e?.Kind())
-			{
-				case SyntaxKind.ImplicitArrayCreationExpression:
-					return ((ImplicitArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ObjectCreationExpression:
-					return ((ObjectCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ArrayCreationExpression:
-					return ((ArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				default:
-					return false;
-			}
-		}
+		static bool ContainsSingleElement(ExpressionSyntax? e) => LinqEnumerableUtils.GetInitializer(e)?.Expressions.Count == 1;
 
 		public static bool LooksLikeAppend(InvocationExpressionSyntax invocation) => (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[1].Expression));
 
 		public static bool LooksLikePrepend(InvocationExpressionSyntax invocation)
 		{
-			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression.TryGetExpressionFromParenthesizedExpression();
 
-			if (expression.IsKind(SyntaxKind.ParenthesizedExpression))
+			if (expression == null)
 			{
-				expression = ((ParenthesizedExpressionSyntax)expression).GetExpression();
+				expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
 			}
 
 			return (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression));
 		}
 
-		public static bool IsList(SemanticModel model, ObjectCreationExpressionSyntax e) => e.Type.ToString().StartsWith("List", StringComparison.Ordinal) && model.GetTypeInfo(e).Type?.MetadataName == "List`1";
-		public static bool IsIEnumerable(SemanticModel model, ExpressionSyntax e) => model.GetTypeInfo(e).Type?.MetadataName == "IEnumerable`1";
+		public static bool IsMonadicArityCollection (SemanticModel model, ExpressionSyntax e)
+		{
+			var typeSymbol = model.GetTypeInfo(e).Type;
+
+			if (typeSymbol == null)
+			{
+				return false;
+			}
+
+			if (typeSymbol.MetadataName.Length != 0 && typeSymbol.MetadataName[typeSymbol.MetadataName.Length - 1] != '1')
```

Comment:
Can we use `^1` here?

Reply:
No. It's only available in .NET Standard 2.1 and we're using .NET Standard 2.0 - should I put that as an issue or just update it now? [System.Index](https://stackoverflow.com/questions/56091142/predefined-type-system-range-is-not-defined-or-imported)

Reply:
We can't use anything other than .Net Standard 2.0. Current versions of Roslyn doesn't support anything earlier and .Net Framework doesn't support anything later, which means we wouldn't be able to run it within VS.

Reply:
Or maybe we could just fake it...
https://sharplab.io/#v2:EYLgHgbALANALiAlgGxgExAagD4AEBMAjALABQuAzAAQFUDCVA3mVazdQPbABWApgMZwqAWQAUAeR4C4AbQC6VAIYAnZQEoqAXgB8S1TIB6hOQG4yAXzJkAdooC2vAM4AHRf141CVMs1JsqjnDKAK6CVACS1mi8YCxsvv7+lBFRMaKI1kIAborIwbwwVMAcHMhUAGbKHHYAolFaFbmOvGpxiUxtiZZ+7d497ckZQgDivHDi5eXNQumZVMi81gDmcAAWajoADGb9VN3dQA===

Reply:
No objections to that solution. Where should I put this though?